api-quality-gate/scripts/selftest.sh

```
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+# selftest.sh — proves the ruleset behaves, so you don't have to trust it blind.
+# 1. The bad sample MUST trip the custom rules.
+# 2. The good sample MUST pass clean.
+set -uo pipefail
+cd "$(dirname "${BASH_SOURCE[0]}")/.."
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)